### PR TITLE
feat: return new added columns in region server's extension response

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

Bump proto dep.

Return the new physical columns to create logical requests and alter logical requests. This patch also changes how the logical columns in a logical table are ordered. They are ordered by column name now instead of column id, since metasrv don't have the correct column id and would be removed soon by https://github.com/GreptimeTeam/greptimedb/issues/2982

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

File: src/operator/src/insert.rs [hunk 1]
```diff
@@ -468,8 +471,9 @@ impl Inserter {
                     &req.table_name,
                 );
 
-                let request_schema = req.rows.as_ref().unwrap().schema.as_slice();
-                let mut create_table_expr = build_create_table_expr(&table_ref, request_schema)?;
+                let mut request_schema = req.rows.as_ref().unwrap().schema.clone();
+                request_schema.sort_unstable_by(|c1, c2| c1.column_name.cmp(&c2.column_name));
+                let mut create_table_expr = build_create_table_expr(&table_ref, &request_schema)?;
```
Comment (waynexia) [inline src/operator/src/insert.rs:476]: We may move this logic to `create_logical_tables` procedure to make it more generic

File: src/metric-engine/src/engine.rs [hunk 2]
```diff
@@ -121,23 +122,39 @@ impl RegionEngine for MetricEngine {
         &self,
         region_id: RegionId,
         request: RegionRequest,
-    ) -> Result<AffectedRows, BoxedError> {
+    ) -> Result<RegionHandleResult, BoxedError> {
+        let mut extension_return_value = HashMap::new();
```
Comment (waynexia) [inline src/metric-engine/src/engine.rs:126]: One unresolved problem is how to merge multiple extensions in batch request. I'm not sure which one is better, making a new batching interface or implement a `merge()` method

File: src/common/meta/src/datanode_manager.rs [hunk 3]
```diff
@@ -41,3 +42,27 @@ pub trait DatanodeManager: Send + Sync {
...
+
+impl HandleResponse {
+    pub fn from_region_response(region_response: RegionResponse) -> Self {
+        Self {
+            affected_rows: region_response.affected_rows as _,
+            extension: region_response.extension,
+        }
+    }
+
+    /// Create one response without extension
```
Comment (fengjiachun) [inline src/common/meta/src/datanode_manager.rs:61]:
```suggestion
    /// Creates one response without extension
```

File: src/common/meta/src/ddl/create_logical_tables.rs [hunk 4]
```diff
@@ -130,7 +131,7 @@ impl CreateLogicalTablesProcedure {
             ));
         }
 
-        // Allocates table ids
+        // Allocates table ids and sort columns on their names.
```
Comment (fengjiachun) [inline src/common/meta/src/ddl/create_logical_tables.rs:134]: Perhaps only comments cannot sort columns.  💯

File: src/metric-engine/src/engine/create.rs [hunk 5]
```diff
@@ -178,6 +190,7 @@ impl MetricEngineInner {
                 metadata_region_id,
                 logical_region_id,
                 new_columns,
+                alter_request,
```
Comment (fengjiachun) [inline src/metric-engine/src/engine/create.rs:193]: I think we need to make the call idempotent, and the `alter_request` should return the same content each time, in order for metasrv to take a retry.

File: src/metric-engine/src/engine/create.rs [hunk 6]
```diff
@@ -51,13 +54,15 @@ impl MetricEngineInner {
         &self,
         region_id: RegionId,
         request: RegionCreateRequest,
+        extension_return_value: &mut HashMap<String, Vec<u8>>,
```
Comment (fengjiachun) [inline src/metric-engine/src/engine/create.rs:57]: Can we use return value? Put it into the params looks not good.

File: src/metric-engine/src/engine/create.rs (hunk 5 above)
Comment (WenyXu) [reply src/metric-engine/src/engine/create.rs:193]: Maybe we should return the physical table schema to metasrv🥲

Comment (fengjiachun) [reply src/metric-engine/src/engine/create.rs:193]: > Maybe we should return the physical table schema to metasrv🥲

+1

**Review by fengjiachun — APPROVED** [review]
LGTM

File: src/metric-engine/src/engine/create.rs (hunk 6 above)
Comment (waynexia) [reply src/metric-engine/src/engine/create.rs:59]: This map will be overwritten in this method. So a mutable map reference is necessary.

**Review by WenyXu — APPROVED** [review]

**Pull request merged**